### ISSUE TITLE
Mono 4.2.1: System.MissingMethodException

### ISSUE THREAD

**Issue by vslepakov** [Azure/amqpnetlite#99]
Version 1.1.0 works as expected the latest version 1.1.6 (and probably a couple before it) throws this exception:

```
System.MissingMethodException: Method 'Socket.get_OSSupportsIPv4' not found.
  at System.Runtime.CompilerServices.AsyncMethodBuilderCore.Start[<ConnectAsync>d__0] (Amqp.<ConnectAsync>d__0& stateMachine) [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1[System.Threading.Tasks.VoidTaskResult].Start[<ConnectAsync>d__0] (Amqp.<ConnectAsync>d__0& stateMachine) [0x00000] in <filename unknown>:0 
  at System.Runtime.CompilerServices.AsyncTaskMethodBuilder.Start[<ConnectAsync>d__0] (Amqp.<ConnectAsync>d__0& stateMachine) [0x00000] in <filename unknown>:0 
  at Amqp.TcpTransport.ConnectAsync (Amqp.Address address, Amqp.ConnectionFactory factory) [0x00000] in <filename unknown>:0 
  at Amqp.TcpTransport.Connect (Amqp.Connection connection, Amqp.Address address, Boolean noVerification) [0x00000] in <filename unknown>:0 
  at Amqp.Connection.Connect (Amqp.Sasl.SaslProfile saslProfile, Amqp.Framing.Open open) [0x00000] in <filename unknown>:0 
  at Amqp.Connection..ctor (Amqp.Address address, Amqp.Sasl.SaslProfile saslProfile, Amqp.Framing.Open open, Amqp.OnOpened onOpened) [0x00000] in <filename unknown>:0 
  at Amqp.Connection..ctor (Amqp.Address address) [0x00000] in <filename unknown>:0 
```

So the latest version is currently not usable with mono.

**Comment by vslepakov** [comment]
OS is Raspbian (on Raspi2)

**Comment by robreeves** [comment]
@vslepakov What version of mono are you running? It seems like you must be running an old version. Socket.OSSupportsIPv4 was added to mono about a year ago. 

https://github.com/mono/mono/commit/dfb71ad5af0105cc8f571fe030247a0e22183c3a#diff-33b28a9dc0b8032650ed77213481a99f

Try upgrading mono.

**Comment by robreeves** [comment]
Maybe I should read the title harder...mono v4.2.1. Checking to see if that property is in the released version now.

**Comment by vslepakov** [comment]
Thanks @robreeves, I am targeting .NET 4.5. I'll try it out with .NET 4.5.2 first thing tomorrow. I appreciate your help

**Comment by vslepakov** [comment]
Ok, the situation is like this: we actually used mono v4.2.1 whole the time but at some point it got downgraded to v4.0.1 (probably the whole system was reinstalled) without me even getting a heads up. So amqpnetlite 1.1.6 does not work with this mono version. Today I upgraded mono to v4.2.2 and also targeted .NET v4.5.2 and everything works as expected. Hence this issue is solved for me.
I am not sure what the minimum mono version you guys claim to support but amqplitenet 1.1.6 in conjunction with  mono 4.0.1 and .NET 4.5 as target **does** have the problem. 
Anyway thank you so much for your help as well as for putting me on the right track
